main.go

```diff
@@ -48,23 +56,71 @@ type Assertion struct {
 
 // Step is
 type Step struct {
-	Name       string      `yaml:"name"`
-	OnNode     int         `yaml:"on_node"`
-	EndNode    int         `yaml:"end_node"`
-	CMD        string      `yaml:"cmd"`
-	Timeout    int         `yaml:"timeout"`
-	Outputs    []Output    `yaml:"outputs"`
-	Inputs     []string    `yaml:"inputs"`
-	Assertions []Assertion `yaml:"assertions"`
+	Name        string      `yaml:"name"`
+
+	/* Old style selection remains supported */
+	OnNode      int         `yaml:"on_node"`
+	EndNode     int         `yaml:"end_node"`
+	/* New style selection */
+	Selection   *Selection   `yaml:"selection"`
+
+	CMD         string      `yaml:"cmd"`
+	Timeout     int         `yaml:"timeout"`
+	Outputs     []Output    `yaml:"outputs"`
+	Inputs      []string    `yaml:"inputs"`
+	Assertions  []Assertion `yaml:"assertions"`
+	WriteToFile string      `yaml:"write_to_file"`
+}
+
+// Selection is
```

Comment:
`Selection is` what ? :) Would be nice to put a small description and the examples from the proposal here (and in the next struct definitions). Note that this get's rendered by Godoc later...